rasa/utils/io.py

```diff
@@ -225,15 +225,69 @@ def unarchive(byte_array: bytes, directory: Text) -> Text:
         return directory
 
 
-def write_yaml_file(data: Dict, filename: Union[Text, Path]) -> None:
+def convert_to_ordered_dict(obj: Dict[Any, Any]) -> OrderedDict:
+    """Convert nested dictionary to an `OrderedDict`.
+
+    Args:
+        obj: Dictionary to convert.
+
+    Returns:
+        An `OrderedDict` with all nested dictionaries converted.
+    """
+    out = OrderedDict()
+
+    for k, v in obj.items():
+        # use recursion on dictionaries
+        if isinstance(v, dict):
+            out[k] = convert_to_ordered_dict(v)
+        # use recursion on lists
+        elif isinstance(v, list):
+            converted_list = []
+            for element in v:
+                # recursion on list elements if they're `dict`s
+                if isinstance(element, dict):
+                    converted_list.append(convert_to_ordered_dict(element))
+                else:
+                    converted_list.append(element)
+            out[k] = converted_list
+        # everything else can just be copied
+        else:
+            out[k] = v
+
+    return out
+
+
+def fix_yaml_representer() -> None:
```

Comment:
optional: `fix` is a bit cryptic to me
```suggestion
def _enable_ordered_dict_yaml_dumping() -> None:
```